### ISSUE TITLE
SSE-KMS - KMS id doesn't exist should throw an error

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#6582]
## Expected Behavior
AWS S3 returns an error when a KMS id doesn't exist. 
```
$ s3cmd put /etc/issue --server-side-encryption-kms-id=02de8641-eea1-4115-8aa2-daf8b1e9b52e s3://vadmeste/obj
upload: '/etc/issue' -> 's3://vadmeste/obj'  [1 of 1]
 26 of 26   100% in    0s    37.84 B/s  done

$ s3cmd put /etc/issue --server-side-encryption-kms-id=02de8641-eea1-4115-8aa2-daf8b1e9b52f s3://vadmeste/obj
upload: '/etc/issue' -> 's3://vadmeste/obj'  [1 of 1]
 26 of 26   100% in    0s    44.89 B/s  done
ERROR: S3 error: 400 (KMS.NotFoundException): Key 'arn:aws:kms:us-east-1:804065449417:key/02de8641-eea1-4115-8aa2-daf8b1e9b52f' does not exist
```

## Current Behavior
Minio server doesn't validate the KMS id, I know that we don't have a concept currently of multiple KMS ids. But we should reject requests if the KMS id is wrong regardless of how many KMS ids we support.
```
$ s3cmd put /etc/issue --no-check-certificate --server-side-encryption-kms-id=02de8641-eea1-4115-8aa2-daf8b1e9b52e s3://vadmeste/obj
upload: '/etc/issue' -> 's3://vadmeste/obj'  [1 of 1]
 26 of 26   100% in    0s   372.35 B/s  done

$  s3cmd put /etc/issue --no-check-certificate --server-side-encryption-kms-id=1 s3://vadmeste/obj
upload: '/etc/issue' -> 's3://vadmeste/obj'  [1 of 1]
 26 of 26   100% in    0s   448.41 B/s  done
```

The actual KMS id is `1`
```
MINIO_SSE_MASTER_KEY=1:a948904f2f0f479b8f8197694b30184b0d2ed1c1cd2a1ec0fb85d299a192a447 minio server ~/test1
```

## Possible Solution
Validate the incoming KMS id

## Steps to Reproduce (for bugs)
Already explained.

## Context
SSE compatibility

## Regression
No, implementation issue. 

## Your Environment
master branch + #6488

**Comment by poornas** [comment]
@harshavardhana , we have actually not implemented  SSE-KMS which follows from https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html. So far, minio server has only implemented SSE-S3 with underlying key management delegated to Vault KMS, or with the MINIO_SSE_MASTER_KEY where operator specified a master key(@aead's PR #6523). However, we should discuss this and improve documentation.

**Comment by harshavardhana** [comment]
> @harshavardhana , we have actually not implemented SSE-KMS which follows from https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html. So far, minio server has only implemented SSE-S3 with underlying key management delegated to Vault KMS, or with the MINIO_SSE_MASTER_KEY where operator specified a master key(@aead's PR #6523). However, we should discuss this and improve documentation.

@poornas I understand that but our API accepts the header with `aws:kms` and but treats it differently, which is not expected from a user point of view.

**Comment by poornas** [comment]
agreed, will send a PR disallowing SSE-KMS headers

**Comment by aead** [comment]
@harshavardhana We don't even have a concept of a single KMS id. SSE-KMS has different ACL semantics - even if there's a KMS id it does not mean an user can use it - SSE-KMS is tied to IAM.
`MINIO_SSE_MASTER_KEY` is **only** for SSE-S3, not SSE-KMS.
The expected behavior for SSE-KMS requests would be `NotImplemented`.

**Comment by harshavardhana** [comment]
> @harshavardhana We don't even have a concept of a single KMS id. SSE-KMS has different ACL semantics - even if there's a KMS id it does not mean an user can use it - SSE-KMS is tied to IAM.
> `MINIO_SSE_MASTER_KEY` is **only** for SSE-S3, not SSE-KMS.
> The expected behavior for SSE-KMS requests would be `NotImplemented`.

That is when you want to create new KMS id's you are allowed to use default KMS id which is used by the SSE-S3 just like KMS - this works without IAM configuration. So I expect that we can honor that. 

For example 

```

$ aws kms list-keys
{
    "Keys": [
        {
            "KeyArn": "arn:aws:kms:us-east-1:804065449417:key/02de8641-eea1-4115-8aa2-daf8b1e9b52e", 
            "KeyId": "02de8641-eea1-4115-8aa2-daf8b1e9b52e"
        }, 
        {
            "KeyArn": "arn:aws:kms:us-east-1:804065449417:key/3c8308ba-9bce-4fc4-b74e-4a2129b83336", 
            "KeyId": "3c8308ba-9bce-4fc4-b74e-4a2129b83336"
        }, 
        {
            "KeyArn": "arn:aws:kms:us-east-1:804065449417:key/b8f75b4e-323f-428e-b55c-7e2073511f14", 
            "KeyId": "b8f75b4e-323f-428e-b55c-7e2073511f14"
        }
    ]
}
```

``3c8308ba-9bce-4fc4-b74e-4a2129b83336`` is our default master key used when SSE-S3 is requested. Same thing can be requested even with kms-id 

```
s3cmd put /etc/issue --no-check-certificate --server-side-encryption-kms-id=3c8308ba-9bce-4fc4-b74e-4a2129b83336 s3://vadmeste/obj
```

```
s3cmd put /etc/issue --no-check-certificate --server-side-encryption s3://vadmeste/obj
upload: '/etc/issue' -> 's3://vadmeste/obj'  [1 of 1]
 26 of 26   100% in    0s    42.54 B/s  done
```

These both are essentially the same

```
$ cat ~/.s3cfg
signature_v2 = False
host_base = s3.amazonaws.com
host_bucket = %(bucket)s.s3.amazonaws.com
bucket_location = us-east-1
use_https = 1
access_key = xxxx
secret_key = xxxx
```

**Comment by aead** [comment]
@harshavardhana The default key (key-id = "") is always accessible by users and can be used by SSE-S3 and SSE-KMS. However the entire idea behind SSE-KMS is that users can specify their own CMKs and attach polices to them - See https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html
So I don't know how useful it is to *partially* implement SSE-KMS (only support an empty client-provided key-ID). I suspect it leads to more uncertainty if users can "use" SSE-KMS (exactly like SSE-S3) but not in the expected/designed way...

**Comment by harshavardhana** [comment]
> @harshavardhana The default key (key-id = "") is always accessible by users and can be used by SSE-S3 and SSE-KMS. However the entire idea behind SSE-KMS is that users can specify their own CMKs and attach polices to them - See https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html
> So I don't know how useful it is to _partially_ implement SSE-KMS (only support an empty client-provided key-ID). I suspect it leads to more uncertainty if users can "use" SSE-KMS (exactly like SSE-S3) but not in the expected/designed way...

I am not talking about KMS API implementation to manage keys or even policies on them - but simply allow KMS-id to be supported just another header. This actually directly translates to 

```
MINIO_SSE_MASTER_KEY=<key_id>:<sha256_hex>
```

Here the `key_id` is `aws_key_id` - in future if we want to implement full KMS we will have a way to handle it. We have no reason to throw NotImplemented since its trivial AFAICS. Since there is no way to create new keys - we simply return an error saying KeyNotFound. 

I would like this style to be considered since it doesn't contradict anything else in implementation.

**Comment by aead** [comment]
@harshavardhana Yeah, this would work and would be quite simple. However from a security perspective this would not add anything. So AFAICS we can do this from an API point of view. However it doesn't change anything if the user can specify which key is used if there is no rule-based key-management in place. So SSE-KMS would exactly be like SSE-S3 but the users must specify the key-ID and if the key-ID is changed the object becomes unavailable...

**Comment by harshavardhana** [comment]
This is fixed by #6587 - we don't support AWS KMS style encryption headers anymore.
